docs/langref/relay/intro.rst

```diff
@@ -0,0 +1,205 @@
+
+## Node
+
+The fundmental unit of the IR is the node, all nodes must have
+a type field.
+
+.. code-block:: python
+    class Node:
+        def checked_type() -> Type:
+            ...
+
+==================
+Variables
+==================
+
+Relay has three variable references local, global, and operators. Our design draws inspiration
+from LLVM which differentiates between identifier types. This enables writers of
+optimizations to know precisely what an identifier references without needing information
+beyond the type of identifier.
+
+Globals are written with `@`, locals are written with `%`, and operators are written with
+no sigil like LLVM's intrinsics. The distinction between global and local identifiers
+makes certain kinds of transformation easier. For example inlining a global definition
+requires no analysis, you can write a pass that just directly inlines the defintions.
+It also ensures there are no spooky action at a distance, introducing a new identifier
+of any type will never introduce an ambiguity to the program.
+
+
+Global Variable
+~~~~~~~~~~~~~~~~~~
+
+Global identifiers are prefixed by the ::@ sigil. A global identifier always
+references a globally visibly definition contained in the environment. You
+can write a global identifier as ::@global.
+
+Local Variable
+~~~~~~~~~~~~~~~~~
+
+Local identifiers are prefixed by the :code:`%` sigil. A local identifier always
+references a parameter, or let bound expression. You can write a local
+identifier as :code:`%local`.
+
+
+================
+Global Functions
+================
+
+A definition consists of a name, type parameter, parameters, and an optional return
+type. A Relay definition is similar to a procedure or function in a typical programming
+language, but can also be viewed as a named subgraph.
+
+A definition minimally consists of an identifier :code:`@id`, an empty set of
+parameters, and a body expression contained by curly braces
+
+.. code-block:: python
+
+    def @id() { body }
+
+A definiton may also contain any number of parameters, for example a
+simple function which just adds two tensors
+
+.. code-block:: python
+
+    def @add(%x, %y) { %x + %y }
+
+It is also possible for us to annotate explicit types on definitions, for example
+we can restrict the above definition to only work on certain types
+
+.. code-block:: python
+
+    def @add(%x: Tensor<Float,10, 10>, %y: Tensor<Float, 10, 10>) -> Tensor<Float, 10, 10> {
+        %x + %y
+    }
+
+A parameter is just a pairing of a :py:class:`~tvm.relay.expr.LocalVar` and optional :py:class:`~tvm.relay.type.Type`. They represent
+the formal parameters of functions and definitions, and are written as :code:`%x : T`.
+
+They may only appear in function literals, and definitions, and have no relation
+to parameters in the machine learning.
+
+When the type information is omitted we will attempt to infer a most general type
+for the users. This property is known as generalization, for a definition without
```

Comment:
I think it would be clearer if the comma after "generalization" were a colon (can misinterpret the "for" in the current phrasing)